src/python/pants/backend/jvm/subsystems/zinc.py

```diff
@@ -192,9 +195,28 @@ def zinc(self):
 
   @property
   def dist(self):
-    """Return the distribution selected for Zinc.
+    """Return the `Distribution` selected for Zinc based on execution strategy.
 
-    :rtype: list of str
+    :rtype: pants.java.distribution.distribution.Distribution
+    """
+    underlying_dist = self.underlying_dist
+    if self._execution_strategy != NailgunTaskBase.HERMETIC:
+      # symlink .pants.d/.jdk -> /some/java/home/
+      jdk_home_symlink = os.path.relpath(
+        os.path.join(self._zinc_factory.get_options().pants_workdir, '.jdk'),
+        get_buildroot())
+
+      if os.path.exists(jdk_home_symlink):
+        os.remove(jdk_home_symlink)
+      os.symlink(underlying_dist.home, jdk_home_symlink)
+      return Distribution(home_path=jdk_home_symlink)
+    else:
+      return underlying_dist
+
+  @property
+  def underlying_dist(self):
```

Comment:
src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py line 435 should use `underlying_dist` not `dist`